packages/editor-ui/src/stores/root.store.ts

```diff
@@ -0,0 +1,225 @@
+import { CLOUD_BASE_URL_PRODUCTION, CLOUD_BASE_URL_STAGING, STORES } from '@/constants';
+import type { RootState } from '@/Interface';
+import { setGlobalState } from 'n8n-workflow';
+import { defineStore } from 'pinia';
+import { computed, ref } from 'vue';
+
+const { VUE_APP_URL_BASE_API } = import.meta.env;
+
+export const useRootStore = defineStore(STORES.ROOT, () => {
+	const state = ref({
```

Comment:
If we go for this store-level state approach, the state should probably be a `reactive` object rather than a `ref`, since it will never get reassigned and is easier to access.